app/services/cart_services/remove_item.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module CartServices
+  class RemoveItem
+    def call(item, user)
+      cart_user = item.user
+      return OpenStruct.new(success?: false, flash: 'Cart does not belong to this user!') if user != cart_user
+
+      if item.destroy!
```

Comment:
https://marcgg.com/blog/2020/04/08/rails-update-bang/
Are you sure it will work as you expected?

Reply:
Great article! What made me use the bang method is that I've read about 'Silent Failure', but it seems I did not understand it 😉  . But in this context, I indeed do need it to return false so changed it to plain `destroy`.

Reply:
Yes, without `!` could be a silent failure, but in this case, you check it.
You can use `!` but you will have to catch an errros.
I'm happy that you currently understand the difference :)